portal-parent/ozwillo-portal-front/src/main/resources/public/js/notifications.jsx.js

```diff
@@ -77,18 +70,26 @@ var NotificationTable = React.createClass({
             currentSort.prop = criterion;
             currentSort.dir = sortDirection;
 
-            var n = component.state.n.sort(function (a, b) {
-                if (typeof a[currentSort.prop] == "number") {
-                    return (a[currentSort.prop] - b[currentSort.prop]) * currentSort.dir;
-                } else {
-                    return a[currentSort.prop].localeCompare(b[currentSort.prop]) * currentSort.dir;
-                }
-            });
+            var n = component.state.n.sort(this.sortElement);
             var state = component.state;
             state.n = n;
             state.currentSort = currentSort;
             component.setState(state);
-        };
+        }.bind(this);
+    },
+    sortElement: function(a,b){
+        var currentSort = this.state.currentSort;
+
+        if (typeof a[currentSort.prop] == typeof a[currentSort.prop]) {
+            if (typeof a[currentSort.prop] == "number") {
+                return (a[currentSort.prop] - b[currentSort.prop]) * currentSort.dir;
+            }
+            if (typeof a[currentSort.prop] == "string") {
+                return a[currentSort.prop].localeCompare(b[currentSort.prop]) * currentSort.dir;
```

Comment:
You didn't replace the use of localeCompare (see https://github.com/ozwillo/ozwillo-portal/issues/343#issuecomment-161368999) ?